### PR TITLE
Add consumer ProGuard configuration

### LINKED ISSUE #1767: Add ProGuard to AAR distribution
When moving to shipping an AAR instead of a JAR we should also bundle the ProGuard configuration so people do not have to do that themselves:

http://stackoverflow.com/questions/30201420/how-to-include-a-proguard-configuration-in-my-android-library-aar

### PR DESCRIPTION
**Pull request by skyisle**
Fixes #1767

### CODE REVIEW

File: realm/realm-library/build.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@ android {
         minSdkVersion 9
         targetSdkVersion 23
         project.archivesBaseName = "realm-android"
+        consumerProguardFiles 'consumer-proguard-rules.pro'
```
Comment (cmelchior) [inline realm/realm-library/build.gradle]: I would just call this `proguard-rules.pro`

**Comment by cmelchior** [comment]
Hi @skyisle 
Thanks for adding this. Can you also add a changelog item about this. Something along the lines of `The Realm AAR now also contains the ProGuard configuration.`

**Comment by cmelchior** [comment]
Looks good to me :+1: 

@realm/java

**Comment by dalinaum** [comment]
:+1:

**Comment by zaki50** [comment]
I also confirmed! :+1:

**Pull request merged**